File: librarySystem/Book.cpp
#include <iostream>
#include "Book.h"
#include "Media.h"
#include <sstream>

using namespace std;

 Book::Book(string &cno, string &titl, string &sub, string &auth, string &desc, string &publ, string &cit, string &yr,
 string &ser, string &note) : Media(cno, titl, sub, note), author(auth), description(desc), publisher(publ),city(cit), year(yr), 
 series(ser)
 {
 }


Book::~Book()
{
}

void Book::display() const
{
	cout<<"--------------Book Details-------------------------------------"<<endl;
	Media::display();
	cout<<"Author-----------------"<<this->author<<endl;
	cout<<"Description------------"<<this->description<<endl;
	cout<<"Publisher--------------"<<this->publisher<<endl;
	cout<<"City-------------------"<<this->city<<endl;
	cout<<"Year-------------------"<<this->year	<<endl;
	cout<<"Series-----------------"<<this->series<<endl;
	cout<<"Notes------------------"<<this->notes<<endl;
	cout<<"---------------------------------------------------------------"<<endl;
}

bool Book::compare_other(const string &other) const
{
	if((this->description.find(other) != std::string::npos) ||
		(this->notes.find(other) != std::string::npos) ||
		(this->year.find(other) != std::string::npos))
	{
		return true;
	}
	else
		return false;
}

File: libPrjPython/Book.py
from Media import Media

class Book(Media):
    
    def __init__(self, callno, title, subject, author, description, publisher, city, year, series, notes):
        super(Book, self).__init__(callno, title, subject, notes)
        self.author = author
        self.description = description
        self.publisher = publisher
        self.city = city
        self.year = year
        self.series = series
    
    def display(self):
       print "---------------Book Details-------------------------------------------------------------------------"
       super(Book, self).display()
       print "Author               : ", self.author
       print "Description          : ", self.description
       print "Publisher            : ", self.publisher
       print "City                 : ", self.city
       print "Year                 : ", self.year
       print "Series               : ", self.series
       print "Notes                : ", self.notes
       print "\n"

    def compare_other(self,srch_other):
        return srch_other in self.description or srch_other in self.notes or srch_other in self.year        
